### PR TITLE
[WIP] support options. 

### PR DESCRIPTION
**Pull request by yamionp**
format, width, height support.

### CODE REVIEW

File: sphinxcontrib/drawio.py [hunk 1]
```diff
@@ -154,16 +177,27 @@ def render_drawio_html(self: HTMLTranslator, node: DrawIONode) -> None:
     if "align" in node:
         self.body.append('<div align="{0}" class="align-{0}">'.format(node["align"]))
 
+    styles = []
+    if "width" in node:
+        styles.append('width:{};'.format(node["width"]))
+    if "height" in node:
+        styles.append('height:{};'.format(node["height"]))
+
+    if len(styles) >= 1:
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py]:
```suggestion
    if styles:
```

File: sphinxcontrib/drawio.py [hunk 2]
```diff
@@ -154,16 +177,27 @@ def render_drawio_html(self: HTMLTranslator, node: DrawIONode) -> None:
     if "align" in node:
         self.body.append('<div align="{0}" class="align-{0}">'.format(node["align"]))
 
+    styles = []
+    if "width" in node:
+        styles.append('width:{};'.format(node["width"]))
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py]:
```suggestion
        styles.append("width:{};".format(node["width"]))
```

File: sphinxcontrib/drawio.py [hunk 3]
```diff
@@ -154,16 +177,27 @@ def render_drawio_html(self: HTMLTranslator, node: DrawIONode) -> None:
...
+    if "height" in node:
+        styles.append('height:{};'.format(node["height"]))
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py]:
```suggestion
        styles.append("height:{};".format(node["height"]))
```

File: tests/conftest.py [hunk 4]
```diff
@@ -6,13 +9,45 @@
...
+    {
+      "drawio_binary_path": "/Applications/draw.io.app/Contents/MacOS/draw.io",
+      "drawio_headless": false
+    }
+    """
+
+    local_conf_path = os.path.join(
+        os.path.dirname(__file__),
+        'roots',
+        'test-root',
```
Comment (Modelmat) [inline tests/conftest.py]: There can be different folders here (aside from just `test-root`, there could be a `test-format` folder).

I would prefer that this work between multiple folders; and actually the formats test be created in a  `test-format` folder (this makes it easier to know which test failed)

File: tests/conftest.py [hunk 5]
```diff
@@ -6,13 +9,45 @@
...
+        'localconf.json',
+    )
+
+    try:
+        pass
```
Comment (Modelmat) [inline tests/conftest.py]:
```suggestion
```

File: tests/conftest.py [hunk 6]
```diff
@@ -6,13 +9,45 @@
...
+        with open(local_conf_path, 'r') as f:
+            d = f.read()
+            conf = json.loads(d)
+            for key, value in conf.items():
+                app.config[key] = value
+                # setattr(app.config, key, value)
```
Comment (Modelmat) [inline tests/conftest.py]:
```suggestion
```

File: tests/conftest.py [hunk 7]
```diff
@@ -6,13 +9,45 @@
...
+        'roots',
+        'test-root',
+        'localconf.json',
+    )
+
+    try:
+        pass
+        with open(local_conf_path, 'r') as f:
+            d = f.read()
+            conf = json.loads(d)
```
Comment (Modelmat) [inline tests/conftest.py]:
```suggestion
            conf = json.loads(d)
            app.config.update(conf)
```

File: tests/conftest.py [hunk 8]
```diff
@@ -21,20 +21,15 @@ def _set_localconf(app):
...
         'localconf.json',
     )
 
     try:
-        pass
         with open(local_conf_path, 'r') as f:
-            d = f.read()
-            conf = json.loads(d)
+            conf = json.load(f)
             for key, value in conf.items():
```
Comment (Modelmat) [inline tests/conftest.py]:
```suggestion
            app.config.update(conf)
```
(and remove the next line)

File: tests/conftest.py (hunk 7 above)
Comment (Modelmat) [reply tests/conftest.py]: `update()`?

Comment (yamionp) [reply tests/conftest.py]: app.config is not dict. It is Config class in sphinx/config.py.
Config class has no update method.

Comment (Modelmat) [reply tests/conftest.py]: My bad.

**Comment by Modelmat** [comment]
Hey @yamionp 
The `draw.io` has a command-line argument for setting the export image size.
I would prefer if this was used rather than the `style` attribute since it would reduce bandwidth.

NB: You'd have to add it to https://github.com/Modelmat/sphinxcontrib-drawio/blob/963ad5d3764a2f0488be77e16ddf9edb97d6fe51/sphinxcontrib/drawio.py#L94 as it changes the actual file output.

**Comment by Modelmat** [comment]
Closing since I've implemented this using proper height/width scaling in #14

**Pull request closed without merging**